src/object/cli_mod.c

```diff
@@ -56,6 +56,10 @@ dc_obj_init(void)
 	if (rc)
 		goto out;
 
+	rc = obj_class_init();
```

Comment:
You need to call obj_class_fini() if the daos_rpc_register() call on L63 fails.

This has caused a regression on master, the log file is at
https://build.hpdd.intel.com/job/daos-stack/job/daos/job/master/3576/artifact/nlt_logs/dnt_cmd_0635_zlzxyw2r.log